pkg/mattermost/mattermost.go

```diff
@@ -232,6 +239,17 @@ func GenerateDeployment(mattermost *mattermostv1alpha1.ClusterInstallation, dbIn
 
 	envVarDB = append(envVarDB, masterDBEnvVar)
 
+	// Add init container to wait for DB setup job to complete
+	initContainers = append(initContainers, corev1.Container{
```

Comment:
This is probably fine where it is, but I would move it down this function until after the minio init containers are possibly attached. My thinking is all database and filestore setup should be completed before we attempt to initialize the mattermost server + database. This obviously won't affect non-minio installations, but better to have this init container go last in all cases.